sql/core/src/test/scala/org/apache/spark/sql/parquet/ParquetIOSuite.scala

```diff
@@ -329,6 +329,16 @@ class ParquetIOSuiteBase extends QueryTest with ParquetTest {
       checkAnswer(parquetFile(file), (data ++ newData).map(Row.fromTuple))
     }
   }
+
+  test("read non-existent parquet file") {
```

Comment:
"SPARK-6330 regression test" can be a better name, so that people can track to the reason why we want to have a test like this.